dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/DataValuesSynchronization.java

```diff
@@ -0,0 +1,243 @@
+
+    private final DataValueSetService dataValueSetService;
+
+    private final SystemSettingManager systemSettingManager;
+
+    private final RestTemplate restTemplate;
+
+
+    @Autowired
+    public DataValuesSynchronization( DataValueService dataValueService, DataValueSetService dataValueSetService, SystemSettingManager systemSettingManager, RestTemplate restTemplate )
+    {
+        this.dataValueService = dataValueService;
+        this.dataValueSetService = dataValueSetService;
+        this.systemSettingManager = systemSettingManager;
+        this.restTemplate = restTemplate;
+    }
+
+    /**
+     * Gets the time of the last successful data synchronization operation. If not set,
+     * the current date subtracted by three days is returned.
+     */
+    private Date getLastDataSynchSuccessFallback()
+    {
+        return (Date) systemSettingManager.getSystemSetting( SettingKey.LAST_SUCCESSFUL_DATA_SYNC );
+    }
+
+    public SynchronizationResult syncDataValuesData()
+    {
+        try
+        {
+            if ( !SyncUtils.testServerAvailability( systemSettingManager, restTemplate ).isAvailable() )
+            {
+                return SynchronizationResult.newFailureResultWithMessage( "DataValues synchronization failed. Remote server is unavailable." );
+            }
+
+            log.info( "Starting DataValues synchronization job." );
+
+            // ---------------------------------------------------------------------
+            // Set time for last success to start of process to make data saved
+            // subsequently part of next synch process without being ignored
+            // ---------------------------------------------------------------------
+
+            final Date startTime = new Date();
+            final Date lastSuccessTime = getLastDataSynchSuccessFallback();
+            final int objectsToSync = dataValueService.getDataValueCountLastUpdatedAfter( lastSuccessTime, true );
+
+            if ( objectsToSync == 0 )
+            {
+                log.debug( "Skipping synchronization, no new or updated DataValues" );
+                return SynchronizationResult.newSuccessResultWithMessage( "Skipping synchronization, no new or updated DataValues" );
+            }
+
+            log.info( "Values: " + objectsToSync + " since last synchronization success: " + lastSuccessTime );
+
+            final String syncUrl = systemSettingManager.getSystemSetting( SettingKey.REMOTE_INSTANCE_URL ) + SyncEndpoint.DATA_VALUE_SETS.getPath();
+            final String username = (String) systemSettingManager.getSystemSetting( SettingKey.REMOTE_INSTANCE_USERNAME );
+            final String password = (String) systemSettingManager.getSystemSetting( SettingKey.REMOTE_INSTANCE_PASSWORD );
+
+            final SystemInstance instance = new SystemInstance( syncUrl, username, password );
+
+            log.info( "Remote server URL for DataValues POST sync: " + instance.getUrl() );
```

Comment:
Just a suggestion: Maybe group up all the basic information logging and write it at the start? This includes: which remote server, page size / values to sync, whether auth was success or not, etc.